core/io/resource_loader.cpp

```diff
@@ -420,6 +420,44 @@ Error ResourceLoader::load_threaded_request(const String &p_path, const String &
 	}
 }
 
+void ResourceLoader::load_threaded_cancel(const String &p_path) {
```

Comment:
If I'm understanding correctly, this is a `load_threaded_get()` without honoring reference counting. In that case, I'd suggest refactoring the code so both functions share a common implementation. There would be a new argument to ignore the ref-count and instead relentessly clear it. Moreover, since it's not actually a proper cancel, I`d name this one `load_threaded_forget()`. But this leads to the question, wouldn't it be even cooler to have a proper `load_threaded_cancel()` that actually stops the load? Nonetheless, that would be much harder to implement and could be an addition for the future. Thoughts?

Reply:
Hello RandomShaper,

Thank you for your feedback. I have read your post and agree with pretty much the entirety of your thoughts. I'm sick right now, but I will get back into this and give a more comprehensive answer.

Refactoring the code is something I had been avoiding as not to break previously existing code, but is obviously worth considering if such a change is positively welcomed by the community.

When it comes to cancelling the actual loading process from load_threaded_request(), I will have to dive deeper into its current implementation and figure out how to do this in a thread-safe manner.

Will spend more time on this when my brain works again.